opentelemetry-sdk/src/opentelemetry/sdk/_metrics/instrument.py

```diff
@@ -86,18 +87,24 @@ def __init__(
         self._measurement_consumer = measurement_consumer
         super().__init__(name, callback, unit=unit, description=description)
 
-        self._callback = callback
+        self._callback: Callable[[], Iterable[APIMeasurement]]
```

Comment:
should this be setting the callback still?

Reply:
I moved it to the else case on L99 so the type annotation is correct (`callback` is a union)